src/participant.rs

```diff
@@ -250,6 +254,16 @@ impl Identity {
         Self::new(verification_key, encryption_key, signature)
             .map_err(|e| io::Error::new(io::ErrorKind::Other, e))
     }
+
+    pub fn verify_data(
```

Comment:
Nit: I would call this `verify()` for symmetry with `sign()`

Reply:
There's already a `verify`, so there would be a duplicate definition. I'm happy to rename the other one or anything else you have in mind.